cache_flat_mutation_reader.hh

```diff
@@ -383,21 +416,27 @@ future<> cache_flat_mutation_reader::read_from_underlying() {
                                     auto it = insert_result.first;
                                     _snp->tracker()->insert(*it);
                                     auto next = std::next(it);
+                                    // Also works in reverse read mode.
+                                    // It preserves the continuity of the range the entry falls into.
                                     it->set_continuous(next->continuous());
                                     clogger.trace("csm {}: inserted empty row at {}, cont={}", fmt::ptr(this), it->position(), it->continuous());
                                 }
                             });
                         } else if (ensure_population_lower_bound()) {
                             with_allocator(_snp->region().allocator(), [&] {
                                 auto e = alloc_strategy_unique_ptr<rows_entry>(
-                                    current_allocator().construct<rows_entry>(*_schema, _upper_bound, is_dummy::yes, is_continuous::yes));
+                                    current_allocator().construct<rows_entry>(table_s, to_table_domain(_upper_bound), is_dummy::yes, is_continuous::no));
```

Comment:
Could you please explain why the entry is no longer continuous?

Reply:
If we're populating in reverse, we haven't yet read the part of the query range which the is_continuous::yes flag marks (for keys smaller than the entry's key). 

The continuity will be set properly a few lines below, immediately:

```
                               if (_read_context.is_reversed()) {
                                    clogger.trace("csm {}: set_continuous({})", fmt::ptr(this), _last_row.position());
                                    _last_row->set_continuous(true);
                                } else {
                                    clogger.trace("csm {}: mark {} as continuous", fmt::ptr(this), insert_result.first->position());
                                    clogger.trace("csm {}: set_continuous({})", fmt::ptr(this), insert_result.first->position());
                                    insert_result.first->set_continuous(true);
                                }
``` 

The second branch makes the behavior in forward mode equivalent to what was before, so we lose nothing.